imports/wasi_snapshot_preview1/wasi_stdlib_test.go

```diff
@@ -48,37 +58,52 @@ var wasmZigCc []byte
 //go:embed testdata/zig/wasi.wasm
 var wasmZig []byte
 
-// wasmGotip is conditionally compiled from testdata/gotip/wasi.go
-var wasmGotip []byte
-
 func Test_fdReaddir_ls(t *testing.T) {
-	for toolchain, bin := range map[string][]byte{
+	toolchains := map[string][]byte{
 		"cargo-wasi": wasmCargoWasi,
+		"tinygo":     wasmTinyGo,
 		"zig-cc":     wasmZigCc,
 		"zig":        wasmZig,
-	} {
+	}
+	if wasmGotip != nil {
+		toolchains["gotip"] = wasmGotip
+	}
+
+	tmpDir := t.TempDir()
+	require.NoError(t, fstest.WriteTestFiles(tmpDir))
+
+	tons := path.Join(tmpDir, "tons")
+	require.NoError(t, os.Mkdir(tons, 0o0777))
+	for i := 0; i < direntCountTons; i++ {
+		require.NoError(t, os.WriteFile(path.Join(tons, strconv.Itoa(i)), nil, 0o0666))
+	}
+
+	for toolchain, bin := range toolchains {
 		toolchain := toolchain
 		bin := bin
 		t.Run(toolchain, func(t *testing.T) {
-			expectDots := toolchain == "zig-cc"
-			testFdReaddirLs(t, bin, expectDots)
+			var expectDots int
+			if toolchain == "zig-cc" {
+				expectDots = 1
+			}
+			testFdReaddirLs(t, bin, toolchain, tmpDir, expectDots)
 		})
 	}
 }
 
-func testFdReaddirLs(t *testing.T, bin []byte, expectDots bool) {
-	// TODO: make a subfs
+const direntCountTons = 8096
+
+func testFdReaddirLs(t *testing.T, bin []byte, toolchain, rootDir string, expectDots int) {
+	t.Helper()
+
 	moduleConfig := wazero.NewModuleConfig().
-		WithFS(fstest.MapFS{
-			"-":   {},
-			"a-":  {Mode: fs.ModeDir},
-			"ab-": {},
-		})
+		WithFSConfig(wazero.NewFSConfig().
+			WithReadOnlyDirMount(path.Join(rootDir, "dir"), "/"))
```

Comment:
So all compilers use our sysfs.DirFS for `ls` tests right now. It is no longer `fs.FS` backed. A later PR will re-add `fs.FS` as a separate dimension, likely masking only `fs.ReadDirFS` to force the zero inode thing.